go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -542,6 +524,58 @@ func (bsnc *BaseSecondaryNetworkController) removePodForSecondaryNetwork(pod *ka
 	return nil
 }
 
+// hasIPAMClaim determines whether a pod's IPAM is being handled by IPAMClaim CR.
+// pod passed should already be validated as having a network connection to nadName
+func (bsnc *BaseSecondaryNetworkController) hasIPAMClaim(pod *kapi.Pod, nadNamespacedName string) (bool, error) {
+	if !bsnc.AllowsPersistentIPs() {
+		return false, nil
+	}
+
+	var ipamClaimName string
+	var wasPersistentIPRequested bool
+	if bsnc.IsPrimaryNetwork() {
+		// primary network ipam reference claim is on the annotation
+		ipamClaimName, wasPersistentIPRequested = pod.Annotations[util.OvnUDNIPAMClaimName]
+	} else {
+		// secondary network the IPAM claim reference is on the network selection element
+		nadKeys := strings.Split(nadNamespacedName, "/")
+		if len(nadKeys) != 2 {
+			return false, fmt.Errorf("invalid NAD name %s", nadNamespacedName)
+		}
+		nadNamespace := nadKeys[0]
+		nadName := nadKeys[1]
+		allNetworks, err := util.GetK8sPodAllNetworkSelections(pod)
+		if err != nil {
+			return false, err
+		}
+		for _, network := range allNetworks {
+			if network.Namespace == nadNamespace && network.Name == nadName {
```

Comment:
I kind of miss the point of this comparison ... 

The `nadNamespacedName` we are passing via input parameter already comes from the pod's network annotations: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/38c2eca51200fe5f7b6a70f7a6604430046006fd/go-controller/pkg/ovn/base_network_controller_secondary.go#L448

I'm trying to make sure iterating the pod's networks is not redundant.

Reply:
what do you propose instead @maiqueb ? We get a list of networks and we need to find which one matches, I don't see a better way. Although I realize i am not breaking the for loop once we find the matching nadName. I'll update that.

Reply:
I though we could get the network-selection-element from the caller, but what we are [iterating](https://github.com/ovn-kubernetes/ovn-kubernetes/blob/38c2eca51200fe5f7b6a70f7a6604430046006fd/go-controller/pkg/ovn/base_network_controller_secondary.go#L448) in the caller method is the podAnnotation, not the network selection elements.

Yeah, you need to do this.